lms/services/canvas_api/client.py

```diff
@@ -210,13 +213,32 @@ def list_files(self, course_id):
         # For documentation of this request see:
         # https://canvas.instructure.com/doc/api/files.html#method.files.api_index
 
-        return self._client.send(
+        files = self._client.send(
             "GET",
             f"courses/{course_id}/files",
             params={"content_types[]": "application/pdf"},
             schema=self._ListFilesSchema,
         )
 
+        # Notify that we've found some files
+        self._request.registry.notify(
+            FilesDiscoveredEvent(
+                request=self._request,
+                values=[
+                    {
+                        "type": "canvas_file",
+                        "course_id": course_id,
+                        "lms_id": file["id"],
+                        "name": file["display_name"],
+                        "size": file["size"],
+                    }
+                    for file in files
+                ],
+            )
+        )
```

Comment:
To be honest I kind of think that this should just to `self._request.db.bulk.upsert(File, values=[...])` and skip out the whole event/subscriber thing. I'm not sure the indirection is adding much